src/screens/login/components/functions/oidc.ts

```diff
@@ -0,0 +1,177 @@
+import type { WebViewNavigation } from 'react-native-webview'
+
+import { getErrorMessage } from '/libs/functions/getErrorMessage'
+import { closeInAppBrowser, showInAppBrowser } from '/libs/intents/InAppBrowser'
+
+const log = Minilog('CozyWebView.functions')
+
+const OIDC_PARAM = 'oidc'
+const USER_CANCELED = 'USER_CANCELED'
+const INVALID_CALLBACK = 'INVALID_CALLBACK'
+
+const OIDC_CALLBACK_URL_PARAM = 'redirect_after_oidc'
+const OIDC_CALLBACK_URL = 'https://links.mycozy.cloud/flagship/oidc_result'
+
+const OIDC_ONBOARD_CALLBACK_URL_PARAM = 'redirect'
+const OIDC_ONBOARD_CALLBACK_URL =
+  'https://links.mycozy.cloud/flagship/oidc_onboarding_finished'
+
+interface OidcLoginCallback {
+  code: string
+  fqdn: string
+  defaultRedirection: string | null
+}
+
+interface OidcOnboardingStartCallback {
+  code: string
+  onboardUrl: string
+}
+
+type OidcCallback = OidcLoginCallback | OidcOnboardingStartCallback
+
+interface OidcOnboardingEndCallback {
+  fqdn: string
+  onboardedRedirection: string | null
+}
+
+/**
+ * Check if the given NavigationRequest should trigger the OIDC scenario
+ */
+export const isOidcNavigationRequest = (
+  request: WebViewNavigation
+): boolean => {
+  try {
+    const url = new URL(request.url)
+
+    const oidcParam = url.searchParams.get(OIDC_PARAM)
+    const isOIDC = oidcParam !== null && oidcParam !== 'false'
+
+    return isOIDC
+  } catch (error) {
+    const errorMessage = getErrorMessage(error)
+    log.error(
+      `Something went wrong while trying to extract OIDC info from URL: ${errorMessage}`
+    )
+    return false
+  }
+}
+
+const parseOIDCOnboardingStartUrl = (
+  url: string
+): OidcOnboardingStartCallback | null => {
+  try {
+    const oidcUrl = new URL(url)
+
+    const code = oidcUrl.searchParams.get('code')
+    const onboardUrl = oidcUrl.searchParams.get('onboard_url')
+
+    if (!code || !onboardUrl) {
+      return null
+    }
+
+    const onboardUrlWithRedirect = new URL(onboardUrl)
+    onboardUrlWithRedirect.searchParams.append(
+      OIDC_ONBOARD_CALLBACK_URL_PARAM,
+      OIDC_ONBOARD_CALLBACK_URL
+    )
+
+    return { code, onboardUrl: onboardUrlWithRedirect.toString() }
+  } catch (error) {
+    const errorMessage = getErrorMessage(error)
+    log.error(
+      `Something went wrong while trying to extract OIDC onboarding info from URL: ${errorMessage}`
+    )
+    return null
+  }
+}
+
+const parseOIDCLoginUrl = (url: string): OidcLoginCallback | null => {
+  try {
+    const oidcUrl = new URL(url)
+
+    const code = oidcUrl.searchParams.get('code')
+    const fqdn = oidcUrl.searchParams.get('fqdn')
+    const defaultRedirection = oidcUrl.searchParams.get('default_redirection')
+
+    if (!code || !fqdn) {
+      return null
+    }
+
+    return { code, fqdn, defaultRedirection }
+  } catch (error) {
+    const errorMessage = getErrorMessage(error)
+    log.error(
+      `Something went wrong while trying to extract OIDC login info from URL: ${errorMessage}`
```

Comment:
same here. We must have error messages on which we can act directly

Reply:
Fixed in https://github.com/cozy/cozy-flagship-app/compare/c351d6a960edac9811994f000ba5a790dcd992bf..02727db31d8c486e4afd0241610cd83f87ebf22a